en/ui-system/components/engine/auto-layout.md

```diff
@@ -2,40 +2,61 @@
+Layout `Type` set to `Grid` will start the grid layout. The grid layout will determine the starting point of the layout based on the combination of `HorizontalDirection` and `VerticalDirection` within a fixed container size, and the layout direction based on the `StartAxis` property.
 
-## Node Direction
+#### Grid Direction
 
-The Layout arrays' child nodes are based on the display order of child nodes in __Hierarchy__ and refers to the array directions set up by the `Vertical Direction` or `Horizontal Direction` properties.
+Layout arranges the child nodes in the order in which they are displayed in the __Hierarchy__, plus the start point and the alignment direction set by the `StartAxis` property.
 
-### Horizontal Direction
+- Start axis
+  - Set to either `HORIZONTAL` or `VERTICAL` orientation. The former will be aligned horizontally, the latter vertically.
+- Start point
+  - The start point is created by combining `HorizontalDirection` and `VerticalDirection`.
+  - Suppose `HorizontalDirection` is `LEFT_TO_RIGHT` and VerticalDirection is `TOP_TO_BOTTOM`, then the start point is __top left__
+  - Suppose `HorizontalDirection` is `RIGHT_TO_LEFT` and VerticalDirection is `BOTTOM_TO_TOP`, then the start point is __bottom right__
 
-You can set up two directions: `Left to Right` or `Right to Left`. The former will array the nodes from left to right according to their display order in __Hierarchy__; the later will array the nodes from right to left according to their display order in __Hierarchy__.
+Two examples are given in conjunction with alignment directions:
 
-### Vertical Direction
+- If the `HorizontalDirection` is set to `LEFT_TO_RIGHT`, `VerticalDirection` is `TOP_TO_BOTTOM` and `StartAxis` is `HORIZONTAL`. This tells the component to be sorted horizontally starting from the __top left__ of the container (below left).
 
-You can set up two directions: `Top to Bottom` or `Bottom to Top`. The former will array the nodes from top to bottom according to their display order in __Hierarchy__; the later will array the nodes from bottom to top according to their display order in __Hierarchy__.
+- If the currently set `HorizontalDirection` is `RIGHT_TO_LEFT`, VerticalDirection is `BOTTOM_TO_TOP` and `StartAxis` is `VERTICAL`, it is telling the component to be sorted vertically starting from the __BOTTOM RIGHT__ of the container (right in the figure below).
 
-## Other layout types are coming soon
+![grid-layout](auto-layout/grid-layout.png)
 
-We will update this part of the document in later edition.
+Grid sorting may also cause the content to exceed the container, which can also be solved by using the `Children` and `Container` modes of `ResizeMode` mentioned in `Horizontal Layout`.
 
-For the properties of other Layout components, please check [Layout](../editor/layout.md) document.
+About Layout components, please refer to the [Layout](../editor/layout.md) documentation.
```

Comment:
About Layout components, please refer to the [Layout](../editor/layout.md) documentation.
-->
for more information, please refer to the [Layout Components](../editor/layout.md) documentation.